### DISCUSSION TITLE
Add functionality to allow some links to be opened in a new tab

### DISCUSSION THREAD

**Discussion by aidopeyto** [GitbookIO/community#577] [Feature Requests]
Recent changes to GitBook mean that all links within a page now open within the same tab, whereas previously it opened all links in a new tab. Normally this is not an issue, but if the Gitbook document is embedded within another website as is an iFrame then the security rules don't allow for the external links to be displayed, showing a connection error instead.

I would like to suggest a feature that would allow an editor to choose between the current tab or new tab, with 'current tab' being the default option to match the current functionality.

**Comment by neiiifellowship** [comment]
likewise!!!

**Comment by firmai** [comment]
This doesn't make any sense, the automated behaviour for external links should be new tab, at the very least make the option available.

**Comment by Frits-Ritense** [comment]
I would like to join the opinion of above 3 participants. Please give us the option to decide where the new page should open.

**Reply by maksim-qd** [reply]
Yes, it's a really big problem for us

**Comment by landoncope** [comment]
+1

**Comment by muchisx** [comment]
Is this possible yet ?

**Comment by Super-Genius** [comment]
I agree, I can't believe they messed this up.  Even adding target="_blank" doesn't work with ```<a href>``` any more.  So ridiculous

**Comment by jaybxyz** [comment]
Is this possible yet?

**Comment by hasanbal** [comment]
still waiting it

**Comment by yoka404** [comment]
++

**Comment by artoritorian77** [comment]
We need it!

**Comment by ShrutiC-git** [comment]
+1

**Comment by AsaEkengren** [comment]
+1

**Comment by SamyPesse** [comment]
At this stage, we don't want to offer this option, as it'll go against the best interest of the end user reading the content.
This is an open debate internally at GitBook, and our position might change in the future, so feedback is more than welcome.

Opening links in new tabs is considered an accessibility issue:
- New tab prevents back navigation from working, forcing users to close the tab to go back
- It is confusing for users with low-vision
- It doesn't match the default browser behavior
- It is not a predictable behavior for the user
- Users have browser mechanisms to open links in a new tab when they want (Command-click, etc)

Some resources about this topic:
- https://www.w3.org/TR/WCAG20-TECHS/G200.html
- https://it.wisc.edu/learn/make-it-accessible/websites-and-web-applications/when-to-open-links-in-a-new-tab/
- https://theadminbar.com/accessibility-weekly/opening-links-in-new-tabs-or-not/


> Normally this is not an issue, but if the Gitbook document is embedded within another website as is an iFrame then the security rules don't allow for the external links to be displayed, showing a connection error instead.

Concerning the case of embedding the page in an iframe, we could consider detecting it in https://github.com/GitbookIO/gitbook/blob/main/packages/gitbook/src/components/primitives/Link.tsx#L53-L59 and opening external links outside the iframe.

Let us know if you feel like we are missing other use cases that should be taken into consideration when evaluating this feature request.

**Reply by SamyPesse** [reply]
I fixed the opening of links when embedded in an iframe in https://github.com/GitbookIO/gitbook/pull/2846

**Reply by MantisClone** [reply]
I strongly disagree with this decision - Gitbook editors should be allowed to select their preferred external link behavior.  My team's immediate response upon reading this is that we must migrate away from Gitbook.

**Reply by MantisClone** [reply]
There are ways to make opening links in a new tab less disorienting, as quoted from the same [article](https://theadminbar.com/accessibility-weekly/opening-links-in-new-tabs-or-not/) you linked:

> the way to improve the accessibility of these links is to warn the user that the link will open in a new tab or window before they click the link. Generally, you do this with a visible icon and hidden screen reader text.

Please give us the option to open links in a new tab and feel free to add a visible icon and hidden screen reader text :pray: 
Please allow this option for external links in the left nav AND for links in the page content :pray:

**Reply by andycloke** [reply]
Please reconsider this. We have many links to external domains from our help docs, and I'd say the UX feels MUCH worse when those aren't in a new tab.

Why not give editors the choice, at least? If people agree with your arguments, they can avoid new tabs.

**Comment by kfc2023** [comment]
maybe consider a popup as an alternative? especially since reusable content is not supported across spaces.  migrating the reader away from the original flow can get them lost. madcap flare has this feature.

**Comment by morsssss** [comment]
I think it would be reasonable to allow new tabs for cases when the link navigates away from the current domain. Many people will find it challenging to get back to the original domain...

**Reply by MarzenaCudecka** [reply]
I agree, it would be great to have an option to choose if it's a new tab for a link or the same one.

**Reply by kcookrose** [reply]
+1

**Comment by SamyPesse** [comment]
Thank you all for the patience! While we still consider that it's not a good practice to have links opening in new tabs and we'll recommend against it; We have re-evaluated our position regarding this feature request, and we have just released a site customization option to configure how **external** links open (internal links / same-domain links will always open in the same tab).

At this stage, we are not considering doing this on a per-link level.

You can find this option in the "Configure" tab of your site customization settings:

<img width="976" alt="Screenshot 2025-06-22 at 15 41 46" src="https://github.com/user-attachments/assets/066923bf-0f39-4bbe-9030-13786e7366ef" />

**Reply by MarzenaCudecka** [reply]
Hey, thank you for making it possible to choose the configuration :)

**Reply by morsssss** [reply]
Yes, thanks! It's always hard to balance the experiences we think users want with the experiences that developers and customers crave. In this case, I think giving us the choice was the right thing to do.

**Reply by aidopeyto** [reply]
As the original creator of this feature request, I would like to thank you for considering this and turning it around into a feature on the platform. It's very much appreciated. It's great to feel listened to 🥹

**Reply by Frits-Ritense** [reply]
Thank you very much @SamyPesse, the setting works like a charm!
We are very pleased with this feature! TY!

**Reply by hellobuhbye** [reply]
Thank you for offering this as an option! This is exactly what my team was looking for.

**Reply by AsaEkengren** [reply]
Very happy about this. Thank you!

**Comment by kfc2023** [comment]
thank you
